include/oneapi/dpl/pstl/zip_view_impl.h

```diff
@@ -0,0 +1,505 @@
+        }
+
+      private:
+        template <std::size_t... In>
+        constexpr bool
+        compare_equal(iterator y, std::index_sequence<In...>) const
+        {
+            return ((std::get<In>(current_) == std::get<In>(y.current_)) || ...);
+        }
+
+        template <typename SentinelsTuple, std::size_t... In>
+        constexpr bool
+        compare_with_sentinels(const SentinelsTuple& sentinels, std::index_sequence<In...>) const
+        {
+            return ((std::get<In>(current_) == std::get<In>(sentinels)) || ...);
+        }
+        
+        template <typename SentinelsTuple, std::size_t... In>
+        constexpr std::common_type_t<std::ranges::range_difference_t<__maybe_const<Const, Views>>...>        
+        distance_to_sentinels(const SentinelsTuple& sentinels, std::index_sequence<In...>) const
+        {
+            return std::ranges::min({difference_type(std::get<In>(current_) - std::get<In>(sentinels))...}, std::less{},
+                                    [](auto a){ return std::abs(a);});
+        }
+        
+        template <std::size_t... In>
+        constexpr std::common_type_t<std::ranges::range_difference_t<__maybe_const<Const, Views>>...>
+        distance_to_it(const iterator it, std::index_sequence<In...>) const
+        {
+            return std::ranges::min({difference_type(std::get<In>(it.current_) - std::get<In>(current_))...}, std::less{},
+                                    [](auto a){ return std::abs(a);});
+        }
+
+        friend class zip_view;
+
+        iterator_type current_;
+    }; // class iterator
+
+    template <bool Const>
+    class sentinel
+    {
+        template <typename... Sentinels>
+        constexpr explicit sentinel(const Sentinels&... sentinels) : end_(sentinels...)
+        {
+        }
+
+      public:
+        sentinel() = default;
+        constexpr sentinel(sentinel<!Const> i) requires Const &&
+            (std::convertible_to<std::ranges::sentinel_t<Views>, std::ranges::sentinel_t<__maybe_const<Const, Views>>> && ...)
+            : end_(std::move(i.end_))
+        {
+        }
+      
+        template <bool OtherConst>
+        requires(std::sentinel_for<std::ranges::sentinel_t<__maybe_const<Const, Views>>, 
+                                   std::ranges::iterator_t<__maybe_const<OtherConst, Views>>>&&...)
+        friend constexpr bool
+        operator==(const iterator<OtherConst>& x, const sentinel& y)
+        {
+            return x.compare_with_sentinels(y.end_, std::make_index_sequence<sizeof...(Views)>());
+        }
+        
+        template <bool OtherConst>
+        requires(std::sized_sentinel_for<std::ranges::sentinel_t<__maybe_const<Const, Views>>, 
+                                         std::ranges::iterator_t<__maybe_const<OtherConst, Views>>>&&...)
+        friend constexpr std::common_type_t<std::ranges::range_difference_t<__maybe_const<Const, Views>>...>        
```

Comment:
Can we use `iterator::difference_type` or make a difference_type visible here to avoid defining the same multiple times.

Reply:
it seems it doesn't work here, for sentinel friend methods..

Reply:
Or probably it makes sense to write simpler - return `auto`?

Reply:
Update: 
In the end, done for iterator and sentinel. CI is green.